### DISCUSSION TITLE
feat: support kwargs to Select.where(), similar to .filter_by()

### DISCUSSION THREAD

**Discussion by wbolster** [sqlalchemy/sqlalchemy#11117] [Ideas]
### Describe the use case

in sqlalchemy 2.x, the `Select` construct is the spiritual successor to the 1.x style `Query` construct.

`Select.where()` already existed, and the 1.x style `Query.filter()` and `.filter_by()` methods got added as `Select.filter()` (an alias to `Select.where()`) and `Select.filter_by()`.

this means there are 3 ways to produce `WHERE` clauses, two of which are identical (`.where()` and `.filter()`), and one of which takes `**kwargs` (`.filter_by()`)

if `Select.where()` would also take keyword arguments (currently it only takes `*conditions` varargs), it could supplant the features offered by `.filter_by()`, and become the the canonical way to add `WHERE` clauses, which is a clean and unsurprising API.

(i wouldn't be surprised if others had the same idea, but i couldn't find any prior discussion about this after a quick search)

### Databases / Backends / Drivers targeted

all

### Example Use

``` python
sqlalchemy.select(...).filter_by(name='example')
```

… currently works, but

``` python
sqlalchemy.select(...).where(name='example')
```

… currently does _not_ work. it would be great if it it did 🙃

**Comment by zzzeek** [comment]
OK so there is history here, which...I can't quite identify, however, we at some point had a method that worked this way - it took `*args` and `**kw`.   I can't remember what it was, however, the takeaway was that people were confused by it.   the `*args` takes arguments of one sort and `**kw` works completely differently, people were confused.   so I would not be in favor of merging these together, I think `where()` is for SQL expressions and `filter_by()` is for keyword matching, two cases that work very differently, is how this should stay.

**Reply by wbolster** [reply]
thanks for the, eeeh, oral history 🙃

i see how these are different, and mixing the two styles could look weird, but the same could be said of any combination of `.where()` and `.filter_by()`.

given the massive API changes for 2.x, and PEP484 (typing) and type checkers being a thing these days, do all these argument still hold in 2024?
